Switch to using a fixture for evohome WaterHeater tests

zxdavb opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->

This PR greatly simplifies the tests of the **WaterHeater** entity, by moving setup logic into **conftest.py**:
 - passes in `evohome` as a fixture rather than using `setup_evohome()`: thus, the integration is no longer setup in the test
 - uses a constant for the **WaterHeater** entity ID instead of extracting it as an attr of the entity  via `get_dhw_entity()` (this function is now removed) 

Once this PR is merged, I'll submit other PRs for replacing `get_ctl_entity()` and `get_zone_entity()` with fixtures

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

zxdavb committed:
initial commit

zxdavb committed:
make DHW entity a fixture

zxdavb committed:
remove unneeded fixtures

oyvindwe reviewed (approved):
Looks like a nice refactoring to me. Can't see any issues with this, but I'm not an expert in Python test frameworks.

oyvindwe commented on tests/components/evohome/conftest.py:
```diff
@@ -153,3 +158,32 @@ async def setup_evohome(
+    config: dict[str, str],
+    install: str,
+) -> AsyncGenerator[EvoDHW]:
+    """Return the first heating zone (Climate entity) of the evohome system."""
```
Hmm - I missed this comment first time around. `get_dhw_entity` returns a water heater entity, and not a climate entity?

zxdavb replied:
Ah yes, I see now. 

Nice spot!

zxdavb committed:
doctweak

joostlek commented on tests/components/evohome/conftest.py:
```diff
@@ -153,3 +158,32 @@ async def setup_evohome(
+def get_dhw_entity(hass: HomeAssistant) -> EvoDHW:
+    """Return the WaterHeater entity of the evohome integration."""
+
+    broker: EvoBroker = hass.data[DOMAIN]["broker"]
+
+    if (dhw := broker.tcs.hotwater) is None:
+        pytest.fail("DHW expected, but was not found")
+
+    entity_registry = er.async_get(hass)
+    entity_id = entity_registry.async_get_entity_id(
+        Platform.WATER_HEATER, DOMAIN, dhw._id
+    )
+
+    component: EntityComponent = hass.data.get(Platform.WATER_HEATER)  # type: ignore[assignment]
+    return next(e for e in component.entities if e.entity_id == entity_id)  # type: ignore[return-value]
```
We should not touch the entity anymore after it is created. This currently is fetching the entity from the paltform and looking at the tests we use it to execute function calls. Instead of doing that directly to the entity, we should mutate the entities like we do in real life, with service calls.

We should avoid touching internals like `hass.data` as well

zxdavb committed:
don't touch `hass.data`

zxdavb committed:
refactor water_heater tests

zxdavb commented on tests/components/evohome/conftest.py:
```diff
@@ -150,6 +161,33 @@ async def setup_evohome(
-        assert mock_client.account_info is not None
+        assert evo and evo.account_info is not None
 
+        mock_client.return_value = evo
```
This is a little cheeky. Is it OK to leave it like this?

The only other option is to yield a tuple, `mock_client, evo`, rather than just `mock_client`, but that would require me to refactor other tests.

joostlek replied:
I am not exactly sure what your concern is. As in, what I now see is that you create your own client as a mock, and patched that into the creation of a new client object. I think that is fine.

zxdavb commented:
I would like to take the same approach, with a `zone` fixture, and a `tcs` (controller) fixture.

But I await your expert input for this PR first.

joostlek commented on tests/components/evohome/conftest.py:
```diff
@@ -150,6 +161,33 @@ async def setup_evohome(
         yield mock_client
+
+
+def get_dhw_entity(hass: HomeAssistant, evo: EvohomeClient) -> EvoDHW:
+    """Return the WaterHeater entity of the evohome integration."""
+
+    if (dhw := evo._get_single_tcs().hotwater) is None:
+        pytest.fail("DHW expected, but was not found")
+
+    entity_id = er.async_get(hass).async_get_entity_id(
+        Platform.WATER_HEATER, DOMAIN, dhw._id
+    )
+
+    component: EntityComponent = hass.data.get(Platform.WATER_HEATER)  # type: ignore[assignment]
+    return next(e for e in component.entities if e.entity_id == entity_id)  # type: ignore[return-value]
```
We still need to get rid of this one though

zxdavb replied:
Oh, I though read-only access would be OK.

How can I get the `entity_id` of the DHW, if all I can get is its `id` and `name`, as in:
```python
        self._attr_unique_id = evo_device.dhwId
        self._attr_name = evo_device.name  # is static
```

zxdavb replied:
Oh, I had though reading attrs was OK. I will construct an entity_id instead.

zxdavb committed:
refactor water_heater fixture as entity_id

joostlek commented:
Maybe I am missing something, but why is the water heater entity id a fixture? You'd assume it would be static?

zxdavb commented:
> Maybe I am missing something, but why is the water heater entity id a fixture? You'd assume it would be static?

It _currently_ happens to be static, but it won't be so in future, and I would like to leave it as a fixture (note that the installation fixtures are not contrived: they are actual user systems, albeit anonymized).

The names for the other evohome entities (controllers, heating zones/rooms) are definitely not static, and so I am 'testing' this scheme on DHW before implementing it for those other entity classes.

Also, I plan for future versions of this integration to support multiple concurrent systems (there is some demand for this), so there will need to be a breaking change for these entity IDs. Putting this in now future-proofs the code, and (IMO) enhances readability.

If you wish, I'll implement it as a constant, but - if so - could you indicate if the approach is acceptable for the climate entities?

joostlek commented:
> The names for the other evohome entities (controllers, heating zones/rooms) are definitely not static

I don't see it as a bad thing that they have to change. I don't even mind if you just create a `DHW_ENTITY = "water_heater.something"` constant and use that.

> so there will need to be a breaking change for these entity IDs

If the entities have a unique id, its not a breaking change

> if so - could you indicate if the approach is acceptable for the climate entities

I currently don't have a good enough idea of what the problem is with the climate entities. Like you see this as a problem so there is probably something to look into, but I don't see it right now.

joostlek commented:
Happy to help looking into what you need to get this integration going (although I keep having a million other things to look at :P )

zxdavb committed:
dhw is now a constant

Use new reauth helpers in weatherflow_cloud (#128821)

Use new reauth helpers in wallbox (#128820)

Simplify custom component loading (#128813)

Use new reauth helpers in webostv (#128823)

Use new reauth helpers in whirlpool (#128825)

Handle invalid zeroconf messages in Android TV Remote (#128819)

Use new reauth helpers in yale (#128828)

Update zhong-hong-hvac to 1.0.13 (#128822)

Use new reauth helpers in vicare (#128779)

Auto lower case username for Schlage auth flows (#128730)

Bump plugwise to v1.4.3 (#128773)

Use new reauth helpers in weheat (#128824)

Use new reauth helpers in youtube (#128835)

Use new reauth helpers in yolink (#128834)

Update attrs to 24.2.0 (#126656)

Add Spotify to strict typing (#128846)

Use new reauth helpers in yalexs_ble (#128831)

Use new reauth helpers in withings (#128826)

Add New Music Category for Media Browser (#128147)

Align consumption sensor names in ViCare integration (#127888)

Reduce the size of the Nest event media storage cache (#128855)

Reduce max media items per nest device

Add humidity to KNX climate (#128844)

Use new reauth helpers in yale_smart_alarm (#128836)

Bump google-nest-sdm to 6.1.3 (#128871)

Bump pyTibber to 0.30.3 (#128860)

Remove myself from roomba codeowners (#128858)

Bump habluetooth to 3.6.0 (#128815)

Add audio feature sensors to Spotify (#128785)

Improve entity cached attributes (#128876)

Use runtime_data for Swiss Public Transport (#128369)

* use runtime_data instead of hass.data[<key>]

* fix service response export type

* reduce runtime_data to be just the coordinator

* fix rebase

* fix ruff

* address reviews

* address reviews

* no general core import

* no general config_entries import

* fix also for services

* remove untyped config entry

* remove unneeded cast

Add translations for Netatmo thermostat preset modes (#128890)

Simplify Swiss public transport coordinator (#128891)

Include Z-Wave JS lowSecurityReason in node added websocket message (#128896)

* Propagate lowSecurityReason to FE when adding a zwavejs device insecurely

* update tests

Remove dead code from concord232 (#128907)

Add reconfigure flow to ring integration (#128357)

Co-authored-by: Christopher Fenner <9592452+CFenner@users.noreply.github.com>
Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

Push real binary sensor states to state machine in tests (#128894)

Use STATE_ON/STATE_OFF constants in template test (#128883)

Bump pyopenweathermap to v0.2.1 (#128892)

Add ecobee set_sensors_used_in_climate service (#102871)

* Add set_active_sensors Service

* Remove version bump from service addition commit

* Reviewer suggested changes

* Changed naming to be more clear of functionality

* Adjusted additional naming to follow new convention

* Updated to pass failing CI tests

* Fix typo

* Fix to pass CI

* Changed argument from climate_name to preset_mode and changed service error

* Made loop more clear and changed raised error to log msg

* Fix typo

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* Removed code that was accidentally added back in and fixed mypy errors

* Add icon for service

* Added sensors as attributes and updated tests

* Revert changes made in #126587

* Added tests for remote_sensors and set_sensors_used_in_climate

* Changed back to load multiplatforms (#126587)

* Check for empty sensor list and negative tests for errors raised

* Added tests and fixed errors

* Add hass to class init to allow for device_registry lookup at startup and check for name changed by user

* Added tests to test the new functions

* Simplified code and fixed testing error for simplification

* Added freeze in test

* Fixed device filtering

* Simplified code section

* Maintains the ability to call `set_sensors_used_in_climate` function even is the user changes the device name from the ecobee app or thermostat without needing to reload home assistant.

* Update tests with new functionality. Changed thermostat identifier to a string, since that is what is provided via the ecobee api

* Changed function parameter

* Search for specific ecobee identifier

* Moved errors to strings.json

* Added test for sensor not on thermostat

* Improved tests and updated device check

* Added attributes to _unrecoreded_attributes

* Changed name to be more clear

* Improve error message and add test for added property

* Renamed variables for clarity

* Added device_id to available_sensors to make it easier on user to find it

---------

Co-authored-by: Robert Resch <robert@resch.dev>
Co-authored-by: Erik Montnemery <erik@montnemery.com>

Add Airzone switch entities to zones (#124562)

Add new QNAP QSW uptime timestamp sensor (#122589)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

Remove explicit templating of persistent_notification service data (#128903)

Add Airzone Cloud main zone mode select (#125918)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

Add Airzone Cloud switch entities to zones (#125917)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

Add floor heating device valve positions in Homematic IP Cloud  (#122759)

* Update sensor.py for new FALMOT Sensors

First Integration attemp to support ValvePosition as Sensor for HmIP-FALMOT-C12

* Update sensor.py

* Update sensor.py

* Add Valve Position to FALMOT-C12

* modified: devcontainer

* Service für minimum vale postion hinzugefügt.

* update to services

* Service call optimized

* Add valvePosition to HomematicIP Cloud for Falmot-C12 and show only channels that are connected with an motorized actuator

* Fix some tests

* Add icon for service

* Fix tests, add check for ValveState in icon

* Remove minimum valve service

* REmove minimum valve

* Use list comprehension for devices, support other terminal blocks

* Remove unused constant

* Check correct channel

---------

Co-authored-by: thecem <46648579+thecem@users.noreply.github.com>

Add fan `set_speed` support for Xiaomi Mi Air Purifier 3C (#126870)

Add config flow to local_file (#125835)

* Add config flow to local_file

* Small mods

* Add/fix tests

* Fix

* slug

* Fix strings

* Mod strings

Add firmware update entity to IronOS integration (#123031)

Add diagnostics to Comelit SimpleHome (#128794)

* Add diagnostics to Comelit SimpleHome

* add test

* add missing tests

* introduce SnapshotAssertion

* cleanup

* exclude date based props

Deprecate entity_id template variable in camera services (#128592)

* Deprecate entity_id template variable in camera services

* Update snapshots

* Tiny lang tweak

* Fix translation

---------

Co-authored-by: Franck Nijhof <git@frenck.dev>

Add diagnostics to Vodafone Station (#128923)

* Add diagnostics to Vodafone Station

* cleanup and exclude props based on date

Add update_percentage property to update entity (#128908)

Allow Trend title to be translated (#128926)

Fix description placeholder in fibaro reauth (#128925)

Allow Random title to be translated (#128928)

Bump holidays to 0.59 (#128924)

Remove explicit templating of telegram_bot service data (#128906)

Remove explicit templating of minio service data (#128905)

Remove explicit templating of velbus service data (#128904)

Remove explicit templating of logbook service data (#128902)

Allow Timer title to be translated (#128927)

Fix description placeholder in brunt reauth (#128933)

* Fix description placeholder in brunt reauth

* Update homeassistant/components/brunt/config_flow.py

Co-authored-by: Jan-Philipp Benecke <github@bnck.me>

* Update homeassistant/components/brunt/config_flow.py

Co-authored-by: Jan-Philipp Benecke <github@bnck.me>

---------

Co-authored-by: Jan-Philipp Benecke <github@bnck.me>

Add subscription tier attribute to Twitch integration. (#128870)

* Add subscription tier to Twitch integration.

* Add test for Twitch tiers.  Tests do not currently pass, so this is only theoretical.

* Fix variable type

* Show tier levels as 1,2,3 instead of the raw API values of 1000,2000,3000.

* Make Twitch subscription tier fixtures strings.

* Use proper assertion value for subscription tier test.

Edited on a bus on my phone. 😎

* Update homeassistant/components/twitch/coordinator.py

* Update tests/components/twitch/test_sensor.py

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

Add missing strings for mold indicator (#128205)

* Add missing localization keys for random component configuration

* Add missing localization keys for mold_indicator component configuration

* one_integration_at_a_time

* Fix localization strings for mold_indicator: use direct values instead of non-existing keys

* Fix localization strings for mold_indicator: use direct values instead of non-existing key

* Add missing translations for Mold Indicator helper

* correcting it for hassfest

* Fixes

---------

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

Drop not needed reauth strings in tplink (#128937)

Use new reauth helpers in unifi (#128837)

* Use new reauth helpers in unifi

* Apply suggestions from code review

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Update config_flow.py

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

Fix description placeholder in imap reauth (#128940)

Implement new state property for alarm_control_panel which is using an enum (#126283)

* Alarm state from enum

* Fixes

* Set final

* Fix rebase

* Test const

* Fix breaking version

* Fix other for alarm_control_panel

* Fix integrations

* More

* More

* More

* More

* Fix zha

* Replace _attr_state

* Fix alarm_control_panel

* Fix tests

* Fixes

* Mods

* Change some

* More

* More

* More

* Tests

* Last tests

* Return enum

* Fix zha

* Remove not needed check

* Fix wording

* Fix homekit

* Mod prometheus

* Fix mypy

* Fix homekit

* Fix ifttt

Fix description placeholder in transmission reauth (#128938)

Add motion detected binary_sensor for tplink (#127883)

* Add motion binary_sensor for tplink

* Remove strings definition as we have device class that handles this

* Simplify instructions

* Remove mentions about fixture creation and snapshot updates as requested

* re-add newline

Expose tplink temperature sensor as measurement (#128640)

Add state_class=measurement to the temperature sensor, making it available for long-term statistics.

Fix flaky update coordinator test (#128943)

Bump xiaomi-ble to 0.33.0 (#128946)

Update astroid to 3.3.5 (#128948)

Bump yarl to 1.16.0 (#128941)

Bump gcal_sync to 6.2.0 (#128949)

Add snapshot service to image entity (#110057)

* Add service definition for saving snapshot of image entity

* Add service to image

* Add tests for image entity service

* Fix tests

* Formatting

* Add service icon

* Formatting

* Formatting

* Raise home assistant error instead of single log error

* Correctly pass entity id

* Raise exception from existing exception

* Expect home assistant error

* Fix services example

* Add test for templated snapshot

* Correct icon service config

* Set correct type for service template

* Remove unneeded

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* remove template

* fix imports

* Update homeassistant/components/image/__init__.py

* Apply suggestions from code review

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

Add OSO Energy services (#118770)

* Add OSO Energy services

* Fixes after review

* Add tests for OSO Energy water heater

* Fixes after review

* Revert changes for service schema in OSO Energy

* Improve osoenergy unit tests

Fix google tasks todo docstrings (#128978)

Add support for fetching bindkey from Mi cloud (#128394)

Fix zha test RuntimeWarnings (#128975)

Bump aiocomelit to 0.9.1 (#128977)

* Bump aiocomelit to 0.9.1

* remove exception

Bump aiovodafone to 0.6.1 (#128976)

* Bump aiovodafone to 0.6.1

* remove exception

Bump PySwitchBot to 0.51.0 (#128990)

Add limited template to at field for time triggers (#126584)

* Add limited template to at field for time triggers

* fix mypy

* Fix comments

* fix-tests

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

Update aioairzone-cloud to v0.6.8 (#128992)

Bump axis to v63 (#129005)

Bump python-roborock to 2.6.1 (#128804)

Bump lektricowifi to 0.0.43 (#128979)

Use ConfigEntry.runtime_data in gardena_bluetooth (#129000)

Improve template docstring (#128967)

Fix step in presets for generic thermostat (#128922)

Expose scripts with no fields as entities (#123061)

Fix FUNDING.yml to OHF (#129013)

Add Hassio HTTP logs/follow to allowed paths (#126606)

* Add logs/follow to admin paths in hassio.http

* Add tests for logs/follow admin paths in hassio.http

* Add tests for logs/follow admin paths in hassio.http

* Add compress and timeout exclusions for hassio http api

* Fix should_compress usage in hassio/ingress

* Add missing follow exceptions for hassio/http

* Add hassio range header forward for logs endpoints

* Fix test syntax hassio/http

Bump orjson to 3.10.10 (#129015)

changelog: https://github.com/ijl/orjson/compare/3.10.9...3.10.10

Adjust logging level in ModBus (#128980)

Fix issue 127570 in ModBus Component

Remove battery device class from bmw secondary sensor (#128970)

Remove battery device class

Add go2rtc binary config to expose api only on localhost (#129025)

Bump github/codeql-action from 3.26.13 to 3.27.0 (#129019)

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 3.26.13 to 3.27.0.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v3.26.13...v3.27.0)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Powerview migrate scene to string unique_id (#128131)

Bump pyduotecno to 2024.10.1 (#128968)

Bump python bsblan version 0.6.4 (#128999)

Allow configuring WebRTC stun and turn servers (#128984)

* Allow configuring WebRTC stun and turn servers

* Add tests

* Remove class WebRTCCoreConfiguration

Have statistics functions return a meaningful, non-none result even if only one value is available (#127305)

* have statistics functions return a meaningful, non-none result even if only one value is available

* improved code coverage

Add switch platform to the Lektrico integration (#126721)

Fix devolo_home_network devices not reporting a MAC address (#129021)

Bump actions/cache from 4.1.1 to 4.1.2 (#129018)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Remove deprecated channel views attribute from Twitch (#129008)

Use runtime_data in balboa (#129035)

Add `completed` to the wait variable when using triggers (`wait_for_trigger`) (#123427)

* Add support for the wait.completed variable when using wait with triggers

* Remove junk comment

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

Use runtime_data in bang_olufsen (#129037)

Create tests for sense integration (#128418)

* Create tests for sense integration

* Rearrange files

* Update to use snapshots

* Update tests/components/sense/__init__.py

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

* Update tests/components/sense/__init__.py

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

* Update tests/components/sense/test_binary_sensor.py

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

* Update tests/components/sense/test_sensor.py

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

* Add missing imports

---------

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

Bump yt-dlp to 2024.10.22 (#129034)

Bump sensorpush-ble to 1.7.0 (#128951)

changelog: https://github.com/Bluetooth-Devices/sensorpush-ble/compare/v1.6.2...v1.7.0

Fix calculation of attributes in group sensor (#128601)

* Fix calculation of attributes in group sensor

* Fixes

* Fixes

* Make module level function

Fix get_time_zone annotations in dt_util (#129050)

Fix cancellation leaking upward from the timeout util (#129003)

zxdavb committed:
rename symbol

zxdavb commented:
Arghh! I can't believe I did that.

zxdavb commented:
I had to rebase and force-push, I think it is OK, but happy to close and resubmit if required.

zxdavb commented:
> Happy to help looking into what you need to get this integration going (although I keep having a million other things to look at :P )

I am grateful - I must admit I am really struggling to move forward on this because everyone is so busy!

zxdavb committed:
add an evohome fixture

zxdavb commented on tests/components/evohome/conftest.py:
```diff
@@ -150,6 +157,21 @@ async def setup_evohome(
         yield mock_client
+
+
+@pytest.fixture
+async def evohome(
+    hass: HomeAssistant,
+    config: dict[str, str],
+    install: str,
+) -> AsyncGenerator[EvohomeClient]:
+    """Return the evohome client for this install fixture."""
+
+    async for mock_client in setup_evohome(hass, config, install=install):
+        evo: EvohomeClient = mock_client.return_value
+
+        yield evo
```
I've had to implement this fixture instead of the `dhw_id` fixture.

I will convert the other tests from using `setup_evohome()`, to this fixture in a future PR.

zxdavb committed:
simplify fixture
